sqlmesh/core/model/definition.py

```diff
@@ -966,14 +966,38 @@ def _data_hash_values(self) -> t.List[str]:
                 statement_exprs = [statement]
             else:
                 rendered = self._statement_renderer(statement).render()
-                if rendered is not None:
+                if self._is_metadata_statement(statement):
+                    continue
+                if rendered:
```

Comment:
For macros that return None, `rendered` is an empty list and will be excluded from the data hash.

Reply:
that’s fine, the args to the func are still included right

Reply:
Yes, statement includes the args to the func.